File: src/main/java/com/wesker926/leetcode/algorithms/p0504/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2022/3/7
// @description: 计算
package s1

import "strconv"

func convertToBase7(num int) string {
	ans := 0
	for i := 1; num != 0; num, i = num/7, i*10 {
		ans += i * (num % 7)
	}
	return strconv.Itoa(ans)
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0504/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0504.s1;

/**
 * @author wesker.gh
 * @date 2022/4/27
 * @description 计算
 */
public class Solution {
    public String convertToBase7(int num) {
        StringBuilder sb = new StringBuilder();
        for (int a = Math.abs(num); a != 0; a /= 7) {
            sb.append(a % 7);
        }
        return num == 0 ? "0" : sb.append(num < 0 ? '-' : "").reverse().toString();
    }
}
